assets/js/pages/ActivityLogPage/ActivityLogPage.jsx

```diff
@@ -52,19 +52,72 @@ const applyDefaultItemsPerPage = (params) =>
     ? params
     : { first: defaultItemsPerPage, ...params };
 
-const activityLogRequest = pipe(
+const activityLogRequestClient = pipe(
   searchParamsToAPIParams,
   applyDefaultItemsPerPage,
   getActivityLog
 );
 
+function MainView({
+  request: { status, response },
+  itemsPerPage,
+  onPageChange,
+  onChangeItemsPerPage,
+}) {
+  if (status === 'initial') {
+    return <> </>;
+  }
+  if (status === 'loading') {
+    return (
+      <div className="flex flex-row min-h-screen justify-center items-center">
+        <Spinner size="xl" />
+      </div>
+    );
+  }
+  if (status === 'failure') {
+    return <div>An error occurred.</div>;
```

Comment:
**suggestion:** we might want to style a tiny bit this case.

<details>
  <summary>See a screenshot here</summary>
  
![image](https://github.com/user-attachments/assets/a33c4e15-c83e-463a-9196-1468eec57328)
</details>

Reply:
I reused the component from the other pages

![image](https://github.com/user-attachments/assets/05b51186-cd2f-44d9-8ba3-6cf55f7a79a3)

Reply:
Awesome, thanks!